Branch-SDK/src/main/java/io/branch/referral/BranchPostTask.java

```diff
@@ -177,7 +177,7 @@ private void onRequestFailed(ServerResponse serverResponse, int status) {
             thisReq_.handleFailure(status, serverResponse.getFailReason());
         }
 
-        if (!thisReq_.shouldRetryOnFail()) {
+        if (status == 400 || !thisReq_.shouldRetryOnFail()) {
```

Comment:
Could we make it a little more generic, like
```
                boolean unretryableErrorCode = (400 <= status && status <= 403);
                 if (unretryableErrorCode || !thisReq_.shouldRetryOnFail())
```

Reply:
Feel free to include any other codes that make sense

Reply:
Sure Benas, let me update the code.